### ISSUE TITLE
unambigious reference to configmaps and secrets

### ISSUE THREAD

**Issue by aklemp** [apache/camel-k#4841]
### Requirement

As a route developer I want to reference properties in existing configmaps and secrets without collision to easier integrate to existing infrastructure and to avoid managing globally unique key names in configmaps and secrets.

### Problem

As [documented](https://camel.apache.org/camel-k/2.0.x/configuration/runtime-properties.html#runtime-configmap) only the key names from configmaps and secrets are used as properties. Obviously, this easily leads to problems: suppose two existing secrets for a source and target system of routes, both containing the keys `username` and `password`. There seems to be no possibility to use this without creating redundant secrets with globally unique key names.

### Proposal

Camel seems to [support](https://camel.apache.org/manual/using-propertyplaceholder.html#_using_kubernetes_property_placeholder_functions) something like `configmap:<name>/<key>[:defaultValue]` and `secret:<name>/<key>[:defaultValue]`. However, trying this, I get the error `Mount path must be configured`. According to mentioned Camel documentation the properties should be available

> Use OS ENV variable with key `camel.k.mount-path.configmaps` and `camel.k.mount-path.secrets` (Camel K compatible).

In my case the environment variable `CAMEL_K_MOUNT_PATH_SECRETS` is set in the pod and the folder contains the secret files as expected.

**Comment by squakez** [comment]
Camel K property setting via `kamel run -p` option is an helper provided in the CLI. If the user has multiple keys with the same value, then, it's something up to the user to know and prevent.

"Using Kubernetes property placeholder functions" is a feature available generically in any runtime using `camel-kubernetes` component. I think that, in order to use it, you need to provide the dependency `-d camel:kubernetes` and the required configuration for the application to connect to the cluster [1] via properties.

However, from a deployment point of view, this may not be the best scenario when managing the application via Camel K operator. Going through that scenario, the application would require to connect to Kubernetes via REST API every time it uses the value, instead of using the native mechanics you can get mounting the configmap/secrets via the operator.

[1] https://camel.apache.org/manual/using-propertyplaceholder.html#_configuring_kubernetes_client

**Comment by aklemp** [comment]
I'm not into the details who is doing its magic (Camel, Camel K, Kubernetes). However, I don't see the need to access REST API every time we use the secret because they are mounted in the pods file system in a directory containing also the secret name. The files containing the secrets can be uniquely references in the file system. There "just" needs to be some format to specify it in the integration and map that to the file contents.

For example `{{secret:<name>/<key>}}` maps to the content of file `CAMEL_K_MOUNT_PATH_SECRETS/<name>/<key>`. From my understanding that is populated on pod startup and never updates even if the secret changes, so the secrets could also be cached in the Camel process of the pod.

**Comment by lburgazzoli** [comment]
According to the [code](https://github.com/apache/camel/blob/main/components/camel-kubernetes/src/main/java/org/apache/camel/component/kubernetes/properties/BasePropertiesFunction.java#L88-L95), there may be an issue in the way the path is discovered.

can you try to sue the [jvm trait ](https://camel.apache.org/camel-k/2.0.x/traits/jvm.html) to set the `camel.k.mount-path.configmaps` and the `camel.k.mount-path.secrets` system properties ?

**Comment by aklemp** [comment]
Tried the following (not sure if my trait value is correct).

```
kamel run --trait jvm.config="-Dcamel.k.mount-path.secrets=/etc/camel/conf.d/_secrets" --config=secret:some  props.groovy
```

```
from('timer:props?period=1000')
    .log('{{secret:some/username}}')
```

That still gives "Mount path must be configured".

**Comment by aklemp** [comment]
Did another check and it doesn't matter if I specify the `jvm.config` trait. The resulting pod looks exactly the same and there is no evidence of any impact. Especially the container args don't have an additional setting. The `jvm.enabled` trait doesn't seem to be necessary because I tried `jvm.classpath` and that entry was visible in the container args.

**Comment by aklemp** [comment]
Stupid me. It works if being used correctly (i.e. `jvm.options` instead of `jvm.config`). The value is referenced successful now.

**Comment by lburgazzoli** [comment]
So I guess the issue is that `camel.k.mount-path.secrets` is uses for both properties and env vars, whereas the env var should have been uppercase to match what camel-k injects

**Comment by aklemp** [comment]
For future reference, it is also working in YAML based integrations:
```yaml
spec:
  traits:
    mount:
      configs:
        - secret:some
    jvm:
      options:
        - "-Dcamel.k.mount-path.secrets=/etc/camel/conf.d/_secrets"
```

**Comment by lburgazzoli** [comment]
@aklemp can you please report an issue on Apache Camel here ? https://issues.apache.org/jira/projects/CAMEL

**Comment by aklemp** [comment]
Sure, created https://issues.apache.org/jira/browse/CAMEL-20054

**Comment by squakez** [comment]
The above was fixed in the framework. We can keep it open and verify as soon as we release next camel k runtime.

**Comment by gansheer** [comment]
@squakez @oscerd This needs to be reopen, I don't know why it was closed.

**Comment by gansheer** [comment]
Fixed by camel-k-runtime 3.8.1
